src/lib/pickles/pseudo/pseudo.mli

```diff
@@ -0,0 +1,59 @@
+(* Pseudo *)
+
+module Make (Impl : Snarky_backendless.Snark_intf.Run) : sig
+  type ('a, 'n) t =
+    'n One_hot_vector.T(Impl).t * ('a, 'n) Pickles_types.Vector.t
+
+  val seal : Impl.Field.t -> Impl.Field.t
+
+  val mask :
+       'n One_hot_vector.T(Impl).t
+    -> (Impl.Field.t, 'n) Pickles_types.Vector.t
+    -> Impl.Field.t
+
+  val choose : ('a, 'n) t -> f:('a -> Impl.Field.t) -> Impl.Field.t
+
+  module Degree_bound : sig
+    type nonrec 'n t = (int, 'n) t
+
+    val shifted_pow : crs_max_degree:int -> 'a t -> Impl.Field.t -> Impl.Field.t
+  end
+
+  module Domain : sig
+    val num_shifts : int
+
+    (** Compute the 'shifts' used by the kimchi permutation argument.
+
+        These values are selected deterministically-randomly to appear outside
+        of the domain, so that every choice of [i] and [shift] results in a
+        distinct value for [shift * domain_generator ^ i].
+
+        Note that, for each different domain size, we attempt to use the same
```

Comment:
This implementation detail probably isn't relevant when using this function; it should be fine to leave this part of the comment in the implementation, but omit it from the doc-comment here.

Reply:
Done